### DISCUSSION TITLE
Basic-ssr-lite and basic-ssr examples write `routerScript` before template's DOCTYPE

### DISCUSSION THREAD

**Discussion by zelid** [TanStack/router#393] [General]
Basic-ssr-lite and basic-ssr examples write `routerScript` before template's DOCTYPE.
Expected to replace `<!--app-head-->` inside `template.html`

![image](https://user-images.githubusercontent.com/179463/203419287-c2bd2010-90e0-4a86-98a9-10e29271febd.png)

![image](https://user-images.githubusercontent.com/179463/203419721-ec972415-22a3-4b54-bf98-689ae88daaae.png)

![image](https://user-images.githubusercontent.com/179463/203420408-33fb0818-2bb8-41df-857b-704b1ab10d4b.png)

Used versions:
 "@tanstack/react-router": "0.0.1-beta.25",
 "@tanstack/react-router-devtools": "0.0.1-beta.19"

Should router script indeed replace the `<!--app-head-->` placeholder in SSR?
Or current behavior is normal?

**Comment by tannerlinsley** [comment]
Technically it should stream to the body as soon as it’s available. I believe this is the current functionality is it not?

Tanner Linsley

**Reply by zelid** [reply]
For some reason it is streaming to the very beginning of HTML document, just before the doctype declaration and the `<!--app-head-->` placeholder is not being replaced.
